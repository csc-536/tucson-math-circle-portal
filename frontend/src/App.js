import logo from "./logo.svg";
import "./App.css";
import Registration from "./registration/registration.js"

function App() {
  return (
    <Registration/>
  );
}

export default App;
