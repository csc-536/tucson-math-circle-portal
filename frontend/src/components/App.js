import Login from "../login/login";
import { Route, BrowserRouter as Router } from "react-router-dom";
import Meetings from "../pages/Meetings";
import Container from "@material-ui/core/Container";
import Registration from "../registration/registration";
import AllStudents from "../allStudentsPage/allStudents";
import NavBar from "./NavBar";
import NewMeeting from "../pages/NewMeeting";
import { useMemo, useState } from "react";
import { AuthContext } from "../contexts/AuthContext";
import { isLoggedIn, loggedInRole } from "../utils";
import MeetingInfo from "../pages/MeetingInfo";
import { makeStyles } from "@material-ui/core";
import Logout from "../pages/Logout";

const useStyles = makeStyles((theme) => ({
  container: {
    marginBottom: theme.spacing(3),
  },
}));

function App() {
  const classes = useStyles();
  const [auth, setAuth] = useState({
    userLoggedIn: isLoggedIn(),
    role: loggedInRole(),
  });
  const providerUser = useMemo(() => ({ auth, setAuth }));
  return (
    <AuthContext.Provider value={providerUser}>
      <Router>
        <NavBar />
        <br />
        <br />
        <br />
        <br />
        <Container fixed className={classes.container}>
          <Route exact path="/" component={Login} />
          <Route
            exact
            path="/signup"
            render={() => <Registration update={false} />}
          />
          <Route
            exact
            path="/profile"
            render={() => <Registration update={true} />}
          />
          <Route exact path="/logout" component={Logout} />
          <Route exact path="/new-meeting" component={NewMeeting} />
          <Route exact path="/meetings" component={Meetings} />
          <Route exact path="/meeting" component={MeetingInfo} />
          <Route exact path="/allStudents" component={AllStudents} />
        </Container>
        <br />
        <br />
        <br />
        <br />
      </Router>
    </AuthContext.Provider>
  );
}

export default App;
