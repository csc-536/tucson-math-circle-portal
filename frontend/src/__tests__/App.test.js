import React from "react";
import App from "../components/App";
import { create } from "react-test-renderer";

describe("Should render login page", () => {
  test("testing base app", () => {
    let tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
