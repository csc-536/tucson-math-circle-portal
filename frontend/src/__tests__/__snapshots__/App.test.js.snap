// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first snapshot test testing app button 1`] = `
Array [
  <div
    className="makeStyles-root-2"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <a
          href="/meetings"
          onClick={[Function]}
        >
          <button
            aria-describedby={null}
            aria-haspopup="true"
            aria-label="home"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeEnd"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="All Meetings"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </a>
        <h5
          className="MuiTypography-root makeStyles-title-3 MuiTypography-h5"
        >
          Tucson Math Circle
        </h5>
        
        
      </div>
    </header>
  </div>,
  <br />,
  <br />,
  <br />,
  <br />,
  <div
    className="MuiContainer-root makeStyles-container-1 MuiContainer-fixed MuiContainer-maxWidthLg"
  >
    <form
      id="loginForm"
      onSubmit={[Function]}
    >
      <h1>
        Login
      </h1>
      <label>
        Email:
        <input
          name="email"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label>
        Password:
        <input
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </label>
      <div>
        <p
          id="fgtPasswrd"
        >
          Forget password?
        </p>
        <p
          id="signUp"
          onClick={[Function]}
        >
          Signup
        </p>
      </div>
      <input
        id="login"
        type="submit"
        value="Login"
      />
    </form>
  </div>,
  <br />,
  <br />,
  <br />,
  <br />,
]
`;

exports[`Should render login page testing base app 1`] = `
Array [
  <div
    className="makeStyles-root-2"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <a
          href="/meetings"
          onClick={[Function]}
        >
          <button
            aria-describedby={null}
            aria-haspopup="true"
            aria-label="home"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeEnd"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="All Meetings"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </a>
        <h5
          className="MuiTypography-root makeStyles-title-3 MuiTypography-h5"
        >
          Tucson Math Circle
        </h5>
        
        
      </div>
    </header>
  </div>,
  <br />,
  <br />,
  <br />,
  <br />,
  <div
    className="MuiContainer-root makeStyles-container-1 MuiContainer-fixed MuiContainer-maxWidthLg"
  >
    <form
      id="loginForm"
      onSubmit={[Function]}
    >
      <h1>
        Login
      </h1>
      <label>
        Email:
        <input
          name="email"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label>
        Password:
        <input
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </label>
      <div>
        <p
          id="fgtPasswrd"
        >
          Forget password?
        </p>
        <p
          id="signUp"
          onClick={[Function]}
        >
          Signup
        </p>
      </div>
      <input
        id="login"
        type="submit"
        value="Login"
      />
    </form>
  </div>,
  <br />,
  <br />,
  <br />,
  <br />,
]
`;
